File: Week6/Day4/E2E_basics/steps/wiki_search_steps.py
from behave import when, then


@when(u'I enter {criteria} into the search bar')
def step_impl(context, criteria: str):
    raise NotImplementedError(u'STEP: When I enter apple into the search bar')


@when(u'I select {language} as my language option')
def step_impl(context, language: str):
    raise NotImplementedError(u'STEP: When I select en as my language option')


@when(u'I click the search button')
def step_impl(context):
    raise NotImplementedError(u'STEP: When I click the search button')


@then(u'I should be on a page with the title {title}')
def step_impl(context, title: str):
    raise NotImplementedError(u'STEP: Then I should be on a page with the title Apple - Wikipedia')

File: Week9/Day3/src/test/java/com/revature/steps/WikiSearchSteps.java
package com.revature.steps;

import com.revature.runner.TestRunner;
import io.cucumber.java.en.Then;
import io.cucumber.java.en.When;
import org.junit.Assert;

public class WikiSearchSteps {

    @When("I enter {string} into the search bar")
    public void i_enter_into_the_search_bar(String string) {
        TestRunner.wikiHome.sendKeysToSearchbar(string);
    }
    @When("I click the search button")
    public void i_click_the_search_button() {
        TestRunner.wikiHome.clickSearchButton();
    }
    @Then("I should be redirected to the {string} page")
    public void i_should_be_redirected_to_the_page(String string) {
        Assert.assertEquals(string, TestRunner.driver.getTitle());
    }
}
